### ISSUE TITLE
Unable to deploy in tomcat from Intellij saying : The Module Root specified is not a module according to Intellij

### ISSUE THREAD

**Issue by vsreekumargit** [zengkid/SmartTomcat#101]
**Describe the bug**
After installing the plugin while trying to run the application, the run is interrupted with the below trace. The project is successfully build using maven and war is also generated but unable to deploy
java.lang.RuntimeException: com.intellij.execution.ExecutionException: The Module Root specified is not a module according to Intellij
	at com.poratu.idea.plugins.tomcat.conf.TomcatCommandLineState.createJavaParameters(TomcatCommandLineState.java:155)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:941)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:68)
	at com.intellij.execution.configurations.JavaCommandLineState.getJavaParameters(JavaCommandLineState.java:41)
	at com.intellij.execution.impl.DefaultJavaProgramRunner.patchJavaCommandLineParams(DefaultJavaProgramRunner.java:137)
	at com.intellij.execution.impl.DefaultJavaProgramRunner.lambda$doExecute$3(DefaultJavaProgramRunner.java:127)
	at com.intellij.execution.runners.JavaProgramPatcher.lambda$patchJavaCommandLineParamsUnderProgress$2(JavaProgramPatcher.java:43)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:299)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:442)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressSynchronously$8(CoreProgressManager.java:526)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:84)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:252)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:188)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:175)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:252)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.intellij.execution.ExecutionException: The Module Root specified is not a module according to Intellij
	at com.poratu.idea.plugins.tomcat.conf.TomcatCommandLineState.createJavaParameters(TomcatCommandLineState.java:108)
	... 28 more

**Screenshots**

![Screen Shot 2022-12-01 at 1 53 44 PM](https://user-images.githubusercontent.com/106341569/205136395-cbc77871-fb78-41f4-baeb-5e743e57235e.png)

**Intellij & SmartTomcat Version (Help -> About copy & paste below)**

IntelliJ IDEA 2022.2 (Community Edition)
Build #IC-222.3345.118, built on July 26, 2022
Runtime version: 17.0.3+7-b469.32 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 12.6.1
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 12
Metal Rendering is ON
Non-Bundled Plugins:
    com.karateca.jstoolbox (1.10)
    org.jetbrains.plugins.yaml (222.3345.126)
    com.poratu.idea.plugins.tomcat (4.3.8)
    camtwoTomcatRunner (1.7)

Kotlin: 222-1.7.10-release-334-IJ3345.118

**Comment by yuezk** [comment]
@vsreekumargit Please ensure that the `Deployment directory` specified is the sub-path of one of your IDEA modules.

**Comment by AAdewunmi** [comment]
@vsreekumargit: I encountered the same problem, resulting in same error message.

@yuezk: Your fix isn't clear. Any chance you can provide steps involved in fixing the problem.Thanks

**Comment by tb12315** [comment]
你发的邮件我已经收到。会在近期内查看。

**Comment by AAdewunmi** [comment]
@vsreekumargit I've solved the problem! 

It has to do with the Tomcat catalina file. If you specify the wrong one for your operating system, it will throw the error above. 

So if you are using Intellij on windows -> your Tomcat installion should point to catalina.bat ... on linux / mac, it should be catalina.sh.

Once I used the right file, my Tomcat server fired up 😎

![Screenshot 2023-01-06 at 15 53 52](https://user-images.githubusercontent.com/15172744/211049059-cbb73b15-54a2-4728-8ea5-dc1ec4caf58e.png)
